tests/TestRegistryTest.cpp

```diff
@@ -323,3 +323,36 @@ TEST(TestRegistry, ResetPluginsWorks)
     myRegistry->resetPlugins();
     LONGS_EQUAL(0, myRegistry->countPlugins());
 }
+
+TEST(TestRegistry, listGroupNames)
+{
+    test1->setGroupName("GROUP_1");
+    myRegistry->addTest(test1);
+    test2->setGroupName("GROUP_2");
+    myRegistry->addTest(test2);
+    test3->setGroupName("GROUP_1");
+    myRegistry->addTest(test3);
+
+    myRegistry->listTestGroupNames(*result);
+    SimpleString s = output->getOutput();
+    STRCMP_EQUAL("GROUP_1 GROUP_2 ", s.asCharString());
```

Comment:
There is no one best solution about trailing delimiters. A lot of people rely on them and I was changing my software in the past to gave clients what they wanted.

After all it's easier to add trailing delimiter than remove it so I will change it.